pennylane/interfaces/jax_jit_tuple.py

```diff
@@ -231,13 +206,19 @@ def wrapper(p):
         res = jax.pure_callback(wrapper, shape_dtype_structs, params, vectorized=True)
         return res
 
-    @execute_wrapper.defjvp
+    @partial(execute_wrapper.defjvp, symbolic_zeros=True)
     def execute_wrapper_jvp(primals, tangents):
         # pylint: disable=unused-variable
         params = primals[0]
 
         # Select the trainable params. Non-trainable params contribute a 0 gradient.
-        trainable_params = [t.trainable_params for t in tapes]
+        list_trainable_parameters = [
+            [idx for idx, t in enumerate(tangent) if not isinstance(t, Zero)]
+            for tangent in tangents[0]
+        ]
```

Comment:
I'm not too familiar with what happens behind-the-scenes in JAX. If all tape parameters are marked as trainable, how does it know when to pass symbolic zeros to `tangents`?

Reply:
Symbolic zeros is the opposite of trainable parameters and is the only official way to access argnums (indirectly) in the custom_jvp. So I create new trainable parameters and override the one that were set previously.